python/dask_cudf/dask_cudf/sorting.py

```diff
@@ -96,7 +96,7 @@ def _append_counts(val, count):
     index = lower  # alias; we no longer need lower
     index[mask] = upper[mask]
     rv = combined_vals.iloc[index]
-    return rv.reset_index(drop=True)
+    return rv.reset_index(drop=True)._data
```

Comment:
Why are we returning the ColumnAccessor here?

Reply:
So until now when we pass a DataFrame & `index` to our `DataFrame` constructor we wouldn't reindex the dataframe and also ignore the `index` that is being passed.

```python
>>> import cudf
>>> df1 = cudf.DataFrame({'a':[1, 2, 3]})
>>> cudf.DataFrame(df1)
   a
0  1
1  2
2  3

# On Master branch
>>> cudf.DataFrame(df1, index=['a', 'b', 'c'])
   a
0  1
1  2
2  3


# On this PR
>>> cudf.DataFrame(df1, index=['a', 'b', 'c'])
      a
a  <NA>
b  <NA>
c  <NA>
>>> cudf.DataFrame(df1._data, index=['a', 'b', 'c'])
   a
a  1
b  2
c  3
```

So that is fixed with these changes. Since that issue is fixed, here in the case of this API, we actually want to apply an index to the data without reindexing the data frame, hence to achieve that we will need to pass `ColumnAccessor` here.

Reply:
In general, I'm -1 to any APIs exposing `ColumnAccessor` (even internal ones). Passing `ColumnAccessors` around between our functions will lead to pain if/when we decide to change the internals of `Frame` to use something else. Can we discuss other solutions?

Reply:
Got around this using `map_partitions`, @rjzamora could you aswell have a look into the dask-cudf changes?